packages/headless/src/controllers/insight-interface/insight-interface.ts

```diff
@@ -0,0 +1,74 @@
+import {InsightEngine} from '../../app/insight-engine/insight-engine';
+import {
+  configuration,
+  insightConfiguration,
+  insightInterface,
+  searchHub,
+} from '../../app/reducers';
+import {fetchInterface} from '../../features/insight-interface/insight-interface-actions';
+import {InsightInterfaceState} from '../../features/insight-interface/insight-interface-state';
+import {
+  ConfigurationSection,
+  InsightConfigurationSection,
+  InsightInterfaceSection,
+  SearchHubSection,
+} from '../../state/state-sections';
+import {loadReducerError} from '../../utils/errors';
+import {buildController, Controller} from '../controller/headless-controller';
+
+/**
+ * The `InsightInterface` controller is responsible for retrieving the Insight interface configuration.
+ */
+export interface InsightInterface extends Controller {
+  /**
+   * Fetches the Insight interface configuration.
+   */
+  fetch(): void;
```

Comment:
```suggestion
  fetchConfiguration(): void;
```
Maybe? Just because it's called an InsightInterface, and only having fetch() could be confusing to think you'd fetch an Interface?

I'm sure you already reflected on this, so put the name you think is good.

Reply:
Well, the function is meant to fetch the interface and, with it, a small part of the Insight configuration. I'll keep `fetch` because it is closer to the "insight interface" concept.